operator/pkg/gateway-api/cell.go

```diff
@@ -221,58 +226,131 @@ func checkCRD(ctx context.Context, clientset k8sClient.Clientset, gvk schema.Gro
 	return nil
 }
 
-func checkRequiredCRDs(ctx context.Context, clientset k8sClient.Clientset) error {
+// checkCRDs checks if required and optional CRDs are present in the cluster,
+// returns an error if the required CRDs are not installed, and returns the
+// schema.GroupVersionKind of any optional CRDs that are installed.
+func checkCRDs(ctx context.Context, clientset k8sClient.Clientset, logger *slog.Logger) ([]schema.GroupVersionKind, error) {
 	var res error
-	for _, gvk := range requiredGVK {
+	var presentGVKs []schema.GroupVersionKind
+
+	for _, gvk := range requiredGVKs {
 		if err := checkCRD(ctx, clientset, gvk); err != nil {
 			res = errors.Join(res, err)
 		}
 	}
-	return res
+
+	for _, optionalGVK := range optionalGVKs {
+		if strings.HasSuffix(optionalGVK.Kind, "lists") {
+			continue
+		}
+		if err := checkCRD(ctx, clientset, optionalGVK); err == nil {
+			// note that the .Kind field contains the _resource_ name -
+			// the plural, lowercase version of the name.
+			optionalGVKList := schema.GroupVersionKind{
+				Group:   optionalGVK.Group,
+				Version: optionalGVK.Version,
+				Kind:    optionalGVK.Kind[:len(optionalGVK.Kind)-1] + "lists",
+			}
+			presentGVKs = append(presentGVKs, optionalGVK, optionalGVKList)
+		} else {
+			logger.Debug("CRD is not present, will not handle it", "optionalGVK", optionalGVK.String())
+		}
+	}
+
+	return presentGVKs, res
 }
 
-// registerReconcilers registers the Gateway API reconcilers to the controller-runtime library manager.
-func registerReconcilers(mgr ctrlRuntime.Manager, translator translation.Translator, logger *slog.Logger) error {
-	reconcilers := []interface {
+// registerReconcilers registers Gateway API reconcilers to the controller-runtime library manager.
+// optionalKinds are previously autodetected based on what CRDs are present in the cluster.
+func registerReconcilers(mgr ctrlRuntime.Manager, translator translation.Translator, logger *slog.Logger, installedCRDs []schema.GroupVersionKind) error {
+	requiredReconcilers := []interface {
 		SetupWithManager(mgr ctrlRuntime.Manager) error
 	}{
 		newGatewayClassReconciler(mgr, logger),
-		newGatewayReconciler(mgr, translator, logger),
+		newGatewayReconciler(mgr, translator, logger, installedCRDs),
 		newReferenceGrantReconciler(mgr, logger),
 		newHTTPRouteReconciler(mgr, logger),
 		newGammaHttpRouteReconciler(mgr, translator, logger),
 		newGRPCRouteReconciler(mgr, logger),
-		newTLSRouteReconciler(mgr, logger),
 	}
 
-	for _, r := range reconcilers {
+	for _, r := range requiredReconcilers {
 		if err := r.SetupWithManager(mgr); err != nil {
 			return fmt.Errorf("failed to setup reconciler: %w", err)
 		}
 	}
 
+	// TODO:
```

Comment:
looks like these TODOs are already addressed and can be deleted (or should be handled in this PR?)